test/test_tree_2d_shallowwater_wet_dry.jl

```diff
@@ -0,0 +1,357 @@
+module TestExamples2DShallowWaterWetDry
+
+using Test
+using Trixi
+using TrixiShallowWater
+
+include("test_trixi.jl")
+
+EXAMPLES_DIR = pkgdir(TrixiShallowWater, "examples", "tree_2d_dgsem")
+
+@testset "Shallow Water Wet Dry" begin
```

Comment:
```suggestion
@testset "TreeMesh2D" begin
```
and a similar structure that gives the mesh type as the name for all the other `@testset ... begin` calls in the testing?

Reply:
This depends on how we want to organize the tests. 
We should decide if we want to have all tests for a given mesh type in a single file, or if we want to have separate files for each system of equations. 
In Trixi.jl there is currently a mixture of both. I would suggest that for now we could just have them all in one file and split the tests apart if this becomes too large.

Reply:
Okay, I just got confused because there is also a mixture here, e.g., line 15 of `test_unstructured_2d.jl` has `@testset "UnstructuredMesh2D" begin`. I am okay with keeping it this way.

Reply:
No you were right the way this test is named is not consistent with the other files.
As we had discussed,  I reorganized the testing such that we now group the tests together in one file for each mesh type in order to reduce the amount of files in the test folder (see 92f2fdde25425999ed967e88dffc6ba1a21c05f9).
If these files eventually become too big, we can then still split them apart.